### DISCUSSION TITLE
Module integration troubles

### DISCUSSION THREAD

**Discussion by LorenzoRottigni** [nuxt/scripts#340] [Q&A]
Hello there!
I recently installed the @nuxt/scripts module to embed a Vimeo player into my Nuxt 3 application. However, after running npx nuxi@latest module add scripts, I started experiencing issues that appear to break my application during the build process. Specifically, errors are triggered during the nuxt prepare call, and the following modules are implicated:

- @nuxt/ui
- nuxt-graphql-codegen

Here’s a screenshot of the errors:
![image](https://github.com/user-attachments/assets/c94913ac-4865-4dfb-9c32-94cd15f33f05)

Initially, I suspected the issue was related to @nuxt/ui and nuxt-graphql-codegen. I spent significant time debugging these packages during the build phase but later identified that the root cause appears to involve the require API in Node.js.

Observed Behavior:
It seems that require() no longer directly imports default exports. Instead, it wraps the imported module as follows:
`{ default: importedModule }`
This causes failures in modules expecting the default export directly.

![image](https://github.com/user-attachments/assets/9775c7ec-d6f6-49c7-a39f-45221ad5c190)

I prefer to use @nuxt/scripts since I plan to integrate additional APIs offered by this module in the future. However, I’d like to better understand what’s causing the integration failure and whether a solution exists.

Thank you for your support!

**Comment by LorenzoRottigni** [comment]
Update:

using pnpm instead of yarn correctly builds the project without any error on my local machine but during CI/CD build on Gitlab it throws the worst nodejs error you can receive... "heap out of memory"
![image](https://github.com/user-attachments/assets/346eb86b-236d-4dc1-bb3b-593bea9c94a1)

**Reply by joaopedrodcf** [reply]
I think without a reproduction repo you understanding the issue is gonna be impossible. From what I could read could be just a problem with versions of packages just because you said just changing from yarn to pnpm made the trick in your local machine

**Comment by harlan-zw** [comment]
Closing as outdated — significant module + bundling changes since this was filed and no clear repro of the build error. If you can still reproduce on the latest `@nuxt/scripts`, please open a fresh issue with a minimal repro 🙏
